vscode-dotnet-runtime-library/src/Acquisition/DotnetResolver.ts

```diff
@@ -0,0 +1,223 @@
+        // Unfortunately even with a new API, that might not go in until .NET 10 and beyond, so we have to rely on dotnet --info for now.*/
+
+        if (!hostPath || hostPath === '""')
+        {
+            return '';
+        }
+
+        const infoCommand = CommandExecutor.makeCommand(`"${hostPath}"`, ['--info']);
+        const envWithForceEnglish = process.env;
+        envWithForceEnglish.DOTNET_CLI_UI_LANGUAGE = 'en-US';
+        // System may not have english installed, but CDK already calls this without issue -- the .NET SDK language invocation is also wrapped by a runtime library and natively includes english assets
+        const hostArch = await (this.executor!).execute(infoCommand, { env: envWithForceEnglish, dotnetInstallToolCacheTtlMs: DOTNET_INFORMATION_CACHE_DURATION_MS }, false).then((result) =>
+        {
+            const lines = result.stdout.split('\n').map((line) => line.trim()).filter((line) => (line?.length ?? 0) > 0);
+            // This is subject to change but there is no good alternative to do this
+            const archLine = lines.find((line) => line.startsWith('Architecture:'));
+            if (archLine === undefined)
+            {
+                this.workerContext.eventStream.post(new DotnetUnableToCheckPATHArchitecture(`Could not find the architecture of the dotnet host ${hostPath}. If this host does not match the architecture ${expectedArchitecture}:
+Please set the PATH to a dotnet host that matches the architecture. An incorrect architecture will cause instability for any C# or .NET related applications that rely on this install.`));
+                if (process.env.DOTNET_INSTALL_TOOL_DONT_ACCEPT_UNKNOWN_ARCH === '1')
+                {
+                    return 'unknown'; // Bad value to cause failure mismatch, which will become 'auto'
+                }
+                else
+                {
+                    return '';
+                }
+            }
+            const arch = archLine.split(' ')[1];
+            return arch;
+        });
+
+        return hostArch;
+    }
+
+    public async getSDKs(existingPath: string, requestedArchitecture: string, knownArchitecture: boolean): Promise<IDotnetListInfo[]>
+    {
+        if (!existingPath || existingPath === '""')
+        {
+            return [];
+        }
+
+        const findSDKsCommand = CommandExecutor.makeCommand(`"${existingPath}"`, ['--list-sdks', '--arch', requestedArchitecture]);
+
+        const sdkInfo = await (this.executor!).execute(findSDKsCommand, { dotnetInstallToolCacheTtlMs: DOTNET_INFORMATION_CACHE_DURATION_MS }, false).then(async (result) =>
+        {
+            if (result.status !== '0')
+            {
+                return [];
+            }
+
+            const architectureKnown = knownArchitecture ? true : await this.hostSupportsArchFlag(existingPath, result.stdout);
+            const sdks = result.stdout.split('\n').map((line) => line.trim()).filter((line) => (line?.length ?? 0) > 0);
```

Comment:
A fun way to clear out your `falsey` values is to just do `.filter(Boolean)` instead of `.filter((line) => (line?.length ?? 0) > 0)`

Reply:
That's... awesome and my code will forever be better from you telling me that, thank you ✨